homeassistant/components/rachio/switch.py

```diff
@@ -67,11 +74,24 @@
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the Rachio switches."""
-    # Add all zones from all controllers as switches
+    has_flex_sched = False
     entities = await hass.async_add_executor_job(_create_entities, hass, config_entry)
+    for entity in entities:
+        if isinstance(entity, RachioSchedule):
+            if entity.type == "FLEX":
+                has_flex_sched = True
```

Comment:
```suggestion
        if isinstance(entity, RachioSchedule) and entity.type == "FLEX":
            has_flex_sched = True
            break
```

Also, can we make "FLEX" a constant ?